server/src/FeatureFinder.ts

```diff
@@ -83,53 +101,62 @@ export class FeatureFinder {
   }
 
   private async installAndFindCommandForV1(): Promise<Command | undefined> {
-    if (this.python === 'python') {
-      this.python = 'python3';
-    }
-
     try {
       await this.installLatestDbtRpc();
     } catch (e) {
       console.log('Error while installing dbt-rpc');
       return undefined;
     }
-
     return new DbtRpcCommand(FeatureFinder.DBT_RPC_PARAMS, this.python);
   }
 
-  private async findDbtRpcGlobalVersion(): Promise<DbtVersion | undefined> {
-    return this.findCommandVersion(new DbtRpcCommand([FeatureFinder.VERSION_PARAM]));
+  private async findDbtRpcGlobalVersion(): Promise<DbtVersionInfo | undefined> {
+    return this.dbtProfileType ? this.findCommandVersion(new DbtRpcCommand([FeatureFinder.VERSION_PARAM]), this.dbtProfileType) : undefined;
   }
 
-  private async findDbtGlobalVersion(): Promise<DbtVersion | undefined> {
-    return this.findCommandVersion(new DbtCommand([FeatureFinder.VERSION_PARAM]));
+  private async findDbtGlobalVersion(): Promise<DbtVersionInfo | undefined> {
+    return this.dbtProfileType ? this.findCommandVersion(new DbtCommand([FeatureFinder.VERSION_PARAM]), this.dbtProfileType) : undefined;
   }
 
-  private async findDbtRpcPythonVersion(): Promise<DbtVersion | undefined> {
+  private async findDbtRpcPythonVersion(): Promise<DbtVersionInfo | undefined> {
     return this.findCommandPythonVersion(new DbtRpcCommand([FeatureFinder.VERSION_PARAM], this.python));
   }
 
-  private async findDbtPythonVersion(): Promise<DbtVersion | undefined> {
+  private async findDbtPythonVersion(): Promise<DbtVersionInfo | undefined> {
     return this.findCommandPythonVersion(new DbtCommand([FeatureFinder.VERSION_PARAM], this.python));
   }
 
-  private async findCommandPythonVersion(command: Command): Promise<DbtVersion | undefined> {
-    return this.python ? this.findCommandVersion(command) : undefined;
+  private async findCommandPythonVersion(command: Command): Promise<DbtVersionInfo | undefined> {
+    return this.python && this.dbtProfileType ? this.findCommandVersion(command, this.dbtProfileType) : undefined;
   }
 
-  private async findCommandVersion(command: Command): Promise<DbtVersion | undefined> {
+  private async findCommandVersion(command: Command, dbtProfileType: string): Promise<DbtVersionInfo> {
     const { stdout, stderr } = await FeatureFinder.DBT_COMMAND_EXECUTOR.execute(command);
-    const vFromStderr = FeatureFinder.readVersion(stderr);
-    const vFromStdout = FeatureFinder.readVersion(stdout);
 
-    return vFromStderr ?? vFromStdout;
-  }
-
-  private static readVersion(data: string): DbtVersion | undefined {
-    return (
-      FeatureFinder.readVersionByPattern(data, FeatureFinder.DBT_VERSION_PATTERN) ??
-      FeatureFinder.readVersionByPattern(data, FeatureFinder.DBT_VERSION_PATTERN_LESS_1_1_0)
-    );
+    const installedVersionFromStderr =
```

Comment:
Probably we can parse stdout only if we don't find anything from stderr. And we can extract this process to separate method.